engine/src/main/java/org/pentaho/di/pan/CommandExecutorResult.java

```diff
@@ -0,0 +1,37 @@
+package org.pentaho.di.pan;
+
+public interface CommandExecutorResult {
+
+  public static enum Result {
+    SUCCESS( 0, "Success" ),
+    ERRORS_INVALID_PROJECT( 10, "The project path is invalid, please provide a valid one" );
```

Comment:
Should not have project or path mentioned in kettle.

Reply:
This `Result` ENUM is part of the `CommandExecutorResult` interface.
Pan.java/Kitchen.java need the code/description from `Result` for exit.
That's why the ENUM is part of a interface in kettle.
Not sure how to work around it.
Your input/idea will be highly appreciated.

Reply:
Have the validation method return a string, or throw an exception, and log that instead? 

The validation message for plugins can't come from an enum in kettle...